system-addon/lib/ASRouter.jsm

```diff
@@ -318,11 +318,13 @@ class _ASRouter {
     }
   }
 
-  async clearMessage(target, id) {
-    if (this.state.currentId === id) {
-      await this.setState({currentId: null});
-    }
-    target.sendAsyncMessage(OUTGOING_MESSAGE_NAME, {type: "CLEAR_MESSAGE"});
+  async blockById(idOrIds) {
```

Comment:
Could you add a test for this method? A lot of asserts from the old unit tests went away with the refactoring.

Reply:
Same goes for testing the end state (e.g. `Router.state.blockList.includes("foo")`) rather than implementation details (e.g. `assert.calledOnce(Router._storage.set)`)